go/s3/README.md

```diff
@@ -20,24 +54,333 @@ or gets the S3 bucket object using a custom timeout of 20 seconds.
 
 `go run CustomHTTPClient.go -b BUCKET -o OBJECT [-s] [-t]`
 
-- *BUCKET* name of the bucket (required).
-- *OBJECT* is the name of the object (required).
-- **-s** shows the object, as a string (optional).
-- **-t** gets the object using a custom timout; otherwise it uses a custom HTTP client.
+- *BUCKET* is name of the bucket.
+- *OBJECT* is the name of the object (required)
+- **-s** shows the object, as a string (optional)
+- **-t** gets the object using a custom timout; otherwise it uses a custom HTTP client
+
+The unit test accepts similar values from *config.json*.
+
+### DeleteBucket/DeleteBucket.go
+
+This example deletes a bucket.
+
+`go run DeleteBucket.go -b BUCKET`
+
+- *BUCKET* is the name of the bucket to delete.
+
+The unit test accepts a similar value from *config.json*.
+
+### DeleteBucketPolicy/DeleteBucketPolicy.go
+
+This example removes the policy for a bucket.
+
+`go run DeleteBucketPolicy.go -b BUCKET`
+
+- *BUCKET* is the name of the bucket.
+
+The unit test accepts a similar value from *config.json*.
+
+### DeleteBucketWebsite/DeleteBucketWebsite.go
+
+This example removes the website configuration for a bucket.
+
+`go run DeleteBucketWebsite.go -b BUCKET`
+
+- *BUCKET* is the name of the bucket.
+
+The unit test accepts a similar value from *config.json*.
+
+### DeleteObject/DeleteObject.go
+
+This example deletes an item from a bucket.
+
+`go run DeleteObject -b BUCKET -i ITEM`
+
+- *BUCKET* is the name of the bucket.
+- *ITEM* is the name of the item.
+
+The unit test accepts similar values from *config.json*.
+
+### DeleteObjects/DeleteObjects.go
+
+This example deletes all of the objects in a bucket.
+
+`go run DeleteObjects.go -b BUCKET`
+
+- *BUCKET* is the name of the bucket to purge of objects.
+
+The unit test accepts a similar value from *config.json*.
+
+### DownloadObject/DownloadObject.go
+
+This example downloads a file from a bucket.
+
+`go run DownloadObject.go -b BUCKET -f FILENAME`
+
+- *BUCKET* is the name of the bucket.
+- *FILENAME* is the name of the bucket item to download to a file.
+
+The unit test accepts similar values from *config.json*.
+
+### EncryptOnServerWithKms/EncryptOnServerWithKms.go
+
+This example adds an object to a bucket with encryption based on an AWS Key Management Service (AWS KMS) key.
+
+`go run EncryptOnServerWithKms.go -b BUCKET -o OBJECT -k KEY`
+
+- *BUCKET* is the name of the bucket.
+- *OBJECT* is the name of the file to upload.
+- *KEY* is the AWS KMS key.
+
+The unit test accepts similar values from *config.json*.
+
+### EnforceMD5/EnforceMD5Content.go
+
+This example enforces an MD5 checksum on the object uploaded to a bucket.
+
+`go run EnforceMD5Content.go -b BUCKET -k KEY`
+
+- *BUCKET* is the name of the bucket.
+- *KEY* is the object uploaded to the bucket.
+
+The unit test accepts similar values from *config.json*.
+
+### GeneratePresignedURL/GeneratePresignedURL.go
+
+This example creates a pre-signed URL for a bucket object.
```

Comment:
...**presigned**...for **an Amazon S3** bucket object

Reply:
Fixed

Reply:
Fixed